arraycontext/impl/pytato/__init__.py

```diff
@@ -112,13 +114,13 @@ def to_numpy(self, array):
         return cl_array.get(queue=self.queue)
 
     def call_loopy(self, program, **kwargs):
-        import pyopencl.array as cla
         from pytato.loopy import call_loopy
+        from arraycontext.impl.pyopencl.taggable_cl_array import TaggableCLArray
 
         entrypoint = program.default_entrypoint.name
 
         # thaw frozen arrays
-        kwargs = {kw: (self.thaw(arg) if isinstance(arg, cla.Array) else arg)
+        kwargs = {kw: (self.thaw(arg) if isinstance(arg, TaggableCLArray) else arg)
                   for kw, arg in kwargs.items()}
```

Comment:
I think it'd be wise to allow tagged and untagged arrays, here and below. (Or, at least, complain about untagged ones.)

Reply:
I don't see a good reason to allow untagged arrays. Just made sure that all the arrays had the correct types.